server/repository/src/db_diesel/campaign/campaign_row.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{
+    ChangeLogInsertRow, ChangelogRepository, ChangelogTableName, RepositoryError, RowActionType, 
+    StorageConnection, Upsert,
+};
+use diesel::prelude::*;
+use serde::{Deserialize, Serialize};
+use chrono::NaiveDate;
+
+table! {
+    campaign(id) {
+        id -> Text,
+        name -> Text,
+        start_date -> Nullable<Date>,
+        end_date -> Nullable<Date>,
```

Comment:
I think we should add deleted_datetime and implement soft delete.
We'll still want to associate campaigns with old stock but won't want to see them.
Arguably we can just change the end date to hide it but still think deleted_datetime would be good.